### PR TITLE
Add feature to vote on tags (#240)

### PR DESCRIPTION
**Pull request by gorgobacka**
Votes can be submitted in the following format:
`{entity_id1: {tag1: vote, ...}, ...}`

e.g.: `{'releaseId': {'tag_1': 'upvote', ...}, ...}`

Therefore, I extended the make_tag_request function. The previous functionality is still the same.

I already tested it on test.musicbrainz.org. But I'm not sure how to write a proper test for it.

### CODE REVIEW

**Comment by gorgobacka** [comment]
This PR would also solve: #196.

**Comment by gorgobacka** [comment]
Bump

**Comment by alastair** [comment]
Thanks for the contributions, but I'm sorry I really don't have the time to look through this and merge it. If you're able to use your modifications then I encourage you to use them, but I don't want to make any promises on when I can merge this because I don't want to fail on them.

**Comment by gorgobacka** [comment]
OK, thanks for your reply. I already used this implementation several times.

**Comment by gorgobacka** [comment]
Bump

**Comment by gorgobacka** [comment]
Was there anything missing for including it in v0.7?

File: musicbrainzngs/mbxml.py [hunk 1]
```diff
@@ -775,10 +775,17 @@ def make_tag_request(**kwargs):
...
                 e_xml.set("{%s}id" % NS, e)
                 taglist = ET.SubElement(e_xml, "{%s}user-tag-list" % NS)
-                for tag in tags:
-                    usertag_xml = ET.SubElement(taglist, "{%s}user-tag" % NS)
-                    name_xml = ET.SubElement(usertag_xml, "{%s}name" % NS)
-                    name_xml.text = tag
+                if isinstance(tags, dict):
+                    for tag, vote in tags.items():
+                        usertag_xml = ET.SubElement(taglist, "{%s}user-tag" % NS)
+                        usertag_xml.set('vote', vote)
```
Comment (alastair) [inline musicbrainzngs/mbxml.py:784]: we should raise an error if the `vote` parameter isn't a valid value

File: musicbrainzngs/musicbrainz.py [hunk 2]
```diff
@@ -1259,8 +1259,10 @@ def submit_tags(**kwargs):
     """Submit user tags.
     Takes parameters named e.g. 'artist_tags', 'recording_tags', etc.,
     and of the form:
+    {entity_id1: {tag1: vote, ...}, ...}
```
Comment (alastair) [inline musicbrainzngs/musicbrainz.py:1278]: The documentation should be updated to say what the value of "vote" can be - "upvote", "downvote", or "withdraw". I wonder if it also makes sense to support alternate values for for this? Integers? 1, 0, -1? Booleans? True, False, None?

Comment (gorgobacka) [reply musicbrainzngs/musicbrainz.py:1278]: Thanks for your comments. I will try to update this branch soon.

And I need to check, if the code still works after all those years.

File: musicbrainzngs/mbxml.py (hunk 1 above)
Comment (gorgobacka) [reply musicbrainzngs/mbxml.py:784]: I implemented a ValueError

File: musicbrainzngs/musicbrainz.py (hunk 2 above)
Comment (gorgobacka) [reply musicbrainzngs/musicbrainz.py:1278]: I improved the documentation now. Is it better now?

For the time being, I would stick with the official values for a vote ("upvote", "downvote", "withdraw"). At the moment, I'm not convinced to include a logic to convert boolean of interger values to this specific strings.

**Comment by gorgobacka** [comment]
@alastair It's ready for review.

**Comment by gorgobacka** [comment]
Ping :)

**Comment by gorgobacka** [comment]
I opened this PR four years ago and got very little feedback. Is there any interest in it, @alastair? If not, please let me know and I can close it.

**Comment by gorgobacka** [comment]
I'm very disappointed... I'm closing this PR now.

**Pull request closed without merging**